WireDomain/Tests/WireDomainTests/Repositories/ConversationLabelsRepositoryTests.swift

```diff
@@ -0,0 +1,332 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+@testable import WireAPI
+import WireAPISupport
+@testable import WireDataModel
+import WireDataModelSupport
+@testable import WireDomain
+import XCTest
+
+final class ConversationLabelsRepositoryTests: XCTestCase {
+
+    var sut: ConversationLabelsRepository!
+    var userPropertiesAPI: MockUserPropertiesAPI!
+
+    var stack: CoreDataStack!
+    let coreDataStackHelper = CoreDataStackHelper()
+    let modelHelper = ModelHelper()
+
+    private var conversation1: ZMConversation!
+    private var conversation2: ZMConversation!
+    private var conversation3: ZMConversation!
+
+    var context: NSManagedObjectContext {
+        stack.syncContext
+    }
+
+    override func setUp() async throws {
+        try await super.setUp()
+        /// Batch requests don't work with in-memory store
+        /// so we need to use a persistent store.
+        stack = try await coreDataStackHelper.createStack(inMemoryStore: false)
+        cleanLabelEntity(in: context)
+        setupConversations()
+        userPropertiesAPI = MockUserPropertiesAPI()
+        sut = ConversationLabelsRepository(
+            userPropertiesAPI: userPropertiesAPI,
+            context: context
+        )
+    }
+
+    override func tearDown() async throws {
+        stack = nil
+        userPropertiesAPI = nil
+        sut = nil
+        try await super.tearDown()
```

Comment:
Do you need to tear down the core data stack?

Reply:
fixed in last commit